src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -395,6 +396,88 @@ struct KVAxesPosition {
     uint32_t batch;
     uint32_t seq_len;
 };
+} // anonymous namespace
+
+class CutTailAfterSlice : public ov::pass::MatcherPass {
+public:
+    OPENVINO_MATCHER_PASS_RTTI("npuw::patterns::CutTailAfterSlice");
+    CutTailAfterSlice::CutTailAfterSlice(
+        std::reference_wrapper<std::shared_ptr<ov::Model>> tail_vocab_mm_ref) {
+        auto slice = opp::wrap_type<ov::op::v8::Slice>({opp::any_input(), opp::any_input(),
+                                                        opp::any_input(), opp::any_input(),
+                                                        opp::any_input()});
+        auto matmul = opp::wrap_type<ov::op::v0::MatMul>({slice, opp::any_input()});
+
+        // There are several patterns for matmul we are looking for:
+        // Matmul -> Result
```

Comment:
There is also a pattern
Matmul -> Convert -> Result
@AsyaPronina could you please handle it as well?

Reply:
Fixed!